### PR TITLE
fix(start_planner): remove front of shift start pose from collision check section

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

points front of shift start pose was used for collision check
That was causing the problem of not generating shift pull out path when object is located at behind of ego vehicle
Screenshot from 2024-02-09 21-46-41

Now shift pull out path can be generated even when object is located at behind of ego vehicle
Screenshot from 2024-02-09 21-59-07

## Tests performed

- [x] [PASS TIERIV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/9adaac94-877c-5238-a14b-b8f238c8bfc5?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/src/start_planner_module.cpp [hunk 1]
```diff
@@ -720,27 +720,36 @@ PathWithLaneId StartPlannerModule::extractCollisionCheckSection(
...
-      return motion_utils::findNearestIndex(
-        combined_path.points, collision_check_end_pose->position);
-    } else {
-      return combined_path.points.size() - 1;
-    }
-  });
-  // remove the point behind of collision check end pose
-  combined_path.points.erase(
-    combined_path.points.begin() + collision_check_end_idx + 1, combined_path.points.end());
-  return combined_path;
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/src/start_planner_module.cpp:743]: memo: 
this kept points before all points before collision_check_end_idx including points before start pose

**Review by kosuke55 — APPROVED** [review]
thanks! LGTM!

**Review by shmpwk — APPROVED** [review]
LGTM

**Comment by kyoichi-sugahara** [comment]
confirmed no degression with start_planner related scenario.

**Pull request merged**